code/modules/hydroponics/hydro_tools.dm

```diff
@@ -148,33 +158,89 @@
 		if(2)
 			dat += "<br>It is a robust and vigorous vine that will spread rapidly."
 
+	if(grown_seed.hematophage)
+		dat += "<br>It is a highly specialized hematophage that will only draw nutrients from blood."
+
 	switch(grown_seed.carnivorous)
 		if(1)
-			dat += "<br>It is carniovorous and will eat tray pests for sustenance."
+			dat += "<br>It is carnivorous and will eat tray pests for sustenance."
 		if(2)
 			dat	+= "<br>It is carnivorous and poses a significant threat to living things around it."
 
 	if(grown_seed.parasite)
 		dat += "<br>It is capable of parisitizing and gaining sustenance from tray weeds."
+
 	if(grown_seed.alter_temp)
-		dat += "<br>It will periodically alter the local temperature by [grown_seed.alter_temp] degrees Kelvin."
+		dat += "<br>It will gradually alter the local room temperature to match it's ideal habitat."
 
-	if(grown_seed.consume_gasses)
-		for(var/gas in grown_seed.consume_gasses)
-			dat += "<br>It will remove [gas] from the environment."
+	if(grown_seed.ligneous)
+		dat += "<br>It is a ligneous plant with strong and robust stems."
 
-	if(grown_seed.exude_gasses)
-		for(var/gas in grown_seed.exude_gasses)
-			dat += "<br>It will release [gas] into the environment."
+	if(grown_seed.thorny)
+		dat += "<br>It posesses a cover of sharp thorns."
+
+	if(grown_seed.stinging)
+		dat += "<br>It posesses a cover of fine stingers capable of releasing chemicals on touch."
+
+	if(grown_seed.teleporting)
+		dat += "<br>It posesses a high degree of temporal/spatial instability and may cause spontaneous bluespace disruptions."
+
+	switch(grown_seed.juicy)
+		if(1)
+			dat += "<br>It's fruit is soft-skinned and abudantly juicy."
+		if(2)
+			dat	+= "<br>It's fruit is excesively soft and juicy."
 
 	if(grown_seed.biolum)
 		dat += "<br>It is [grown_seed.biolum_colour ? "<font color='[grown_seed.biolum_colour]'>bio-luminescent</font>" : "bio-luminescent"]."
 	if(grown_seed.flowers)
 		dat += "<br>It has [grown_seed.flower_colour ? "<font color='[grown_seed.flower_colour]'>flowers</font>" : "flowers"]."
 
+	dat += "<br><br>\[<a href='?src=\ref[src];print=1'>print report</a>\] \[<a href='?src=\ref[src];clear=1'>clear</a>\]"
 	if(dat)
 		user << browse(dat,"window=plant_analyzer")
+		last_data = dat
+	return
 
+/obj/item/device/analyzer/plant_analyzer/attack_self(mob/user as mob)
+	if(last_data)
+		user << browse(last_data,"window=plant_analyzer")
+	else
+		user << "<span class='notice'>\icon[src] No plant scan data in memory.</span>"
+	return 0
+
+/obj/item/device/analyzer/plant_analyzer/proc/print_report_verb()
+	set name = "Print Plant Report"
+	set category = "Object"
+	set src = usr
```

Comment:
Er, I'm not experienced with verbs, but shouldn't this be `set src in usr`? I think this'll only make it usable if the player's the analyzer, which... is impossible.